src/cls/rgw/cls_rgw_client.cc

```diff
@@ -773,3 +773,91 @@ int cls_rgw_lc_list(IoCtx& io_ctx, string& oid,
 
  return r;
 }
+
+void cls_rgw_reshard_add(librados::ObjectWriteOperation& op, const cls_rgw_reshard_entry& entry)
+{
+  bufferlist in;
+  struct cls_rgw_reshard_add_op call;
+  call.entry = entry;
+  ::encode(call, in);
+  op.exec("rgw", "reshard_add", in);
+}
+
+int cls_rgw_reshard_list(librados::IoCtx& io_ctx, const string& oid, string& marker, uint32_t max,
+                         list<cls_rgw_reshard_entry>& entries, bool& is_truncated)
```

Comment:
@oritwas nit, we prefer to pass return values such as `is_truncated` here by pointer